src/ifcwrap-csharp/IfcParseWrapper.i

```diff
@@ -0,0 +1,918 @@
+		std::vector<const IfcParse::inverse_attribute*>::const_iterator it = attrs.begin();
+		for (; it != attrs.end(); ++it) {
+			if ((*it)->name() == name) {
+				return 2;
+			}
+		}
+		}
+
+		return 0;
+	}
+
+	// id() is defined on IfcBaseEntity and not on IfcBaseClass, in order
+	// to expose it to the Python wrapper it is simply duplicated here.
+	// Same applies to the two methods reimplemented below.
+	int id() const {
+		return $self->data().id();
+	}
+
+	int Length() const {
+		if ($self->declaration().as_entity()) {
+			return $self->declaration().as_entity()->attribute_count();
+		} else {
+			return 1;
+		}
+	}
+
+	std::vector<std::string> get_attribute_names() const {
+		if (!$self->declaration().as_entity()) {
+			return std::vector<std::string>(1, "wrappedValue");
+		}
+		
+		const std::vector<const IfcParse::attribute*> attrs = $self->declaration().as_entity()->all_attributes();
+		
+		std::vector<std::string> attr_names;
+		attr_names.reserve(attrs.size());		
+		
+		std::vector<const IfcParse::attribute*>::const_iterator it = attrs.begin();
+		for (; it != attrs.end(); ++it) {
+			attr_names.push_back((*it)->name());
+		}
+
+		return attr_names;
+	}
+
+	std::vector<std::string> get_inverse_attribute_names() const {
+		if (!$self->declaration().as_entity()) {
+			return std::vector<std::string>(0);
+		}
+
+		const std::vector<const IfcParse::inverse_attribute*> attrs = $self->declaration().as_entity()->all_inverse_attributes();
+		
+		std::vector<std::string> attr_names;
+		attr_names.reserve(attrs.size());		
+		
+		std::vector<const IfcParse::inverse_attribute*>::const_iterator it = attrs.begin();
+		for (; it != attrs.end(); ++it) {
+			attr_names.push_back((*it)->name());
+		}
+
+		return attr_names;
+	}
+	
+	bool is_a(const std::string& s) {
+		return self->declaration().is(s);
+	}
+
+	std::string is_a(bool with_schema=false) const {
+		auto t = self->declaration().name();
+		if (with_schema) {
+			t = self->declaration().schema()->name() + "." + t;
+		}
+		return t;
+	}
+
+	std::pair<IfcUtil::ArgumentType,Argument*> get_argument(unsigned i) {
+		return std::pair<IfcUtil::ArgumentType,Argument*>($self->data().getArgument(i)->type(), $self->data().getArgument(i));
+	}
+
+	std::pair<IfcUtil::ArgumentType,Argument*> get_argument(const std::string& a) {
+		unsigned i = $self->declaration().as_entity()->attribute_index(a);
+		return std::pair<IfcUtil::ArgumentType,Argument*>($self->data().getArgument(i)->type(), $self->data().getArgument(i));
+	}
+
+	bool Equals(IfcUtil::IfcBaseClass* other) const {
+		return $self->identity() == other->identity();
+	}
+
+	std::string TString() const {
+		return $self->data().toString();
+	}
+
+	std::string ToStringSpf(bool valid_spf) const {
+		return $self->data().toString(valid_spf);
+	}
+
+	// Just something to have a somewhat sensible value to hash
+	size_t file_pointer() const {
+		return reinterpret_cast<size_t>($self->data().file);
+	}
```

Comment:
Consider renaming this method if its only purpose is for hashing

Reply:
i yeeted it instead